### PR TITLE
Pay 3934 payment nightly build

### PR DESCRIPTION
**Pull request by Jennings-HMCTS**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by amitchidrewar1301 — APPROVED** [review]

File: api-contract/build.gradle [hunk 1]
```diff
@@ -1,5 +1,18 @@
 
 description = ''
+configurations.all {
```
Comment (AyanDuttaHMCTS) [inline api-contract/build.gradle:3]: Looks like this is a generic configuration required for all the parent & child modules, if we keep this in parent(ccpay-payment-app) only, will this not work for the child modules(/api, /api-contract, /model etc) as well ?

File: api/build.gradle [hunk 2]
```diff
@@ -34,6 +34,19 @@ sourceSets {
     }
 }
 
+configurations.all {
```
Comment (AyanDuttaHMCTS) [inline api/build.gradle:37]: Looks like this is a generic configuration required for all the parent & child modules, if we keep this in parent(ccpay-payment-app) only, will this not work for the child modules(/api, /api-contract, /model etc) as well ?

File: build.gradle [hunk 3]
```diff
@@ -101,7 +113,15 @@ subprojects {
             exceptionFormat = 'full'
         }
     }
-
+    configurations.all {
+        resolutionStrategy {
+            eachDependency { DependencyResolveDetails det ->
```
Comment (AyanDuttaHMCTS) [inline build.gradle:134]: Are we not duplicating the same configuration in the same file ?

File: api-contract/build.gradle (hunk 1 above)
Comment (Jennings-HMCTS) [reply api-contract/build.gradle:3]: No @AyanDuttaHMCTS , that doesnt work. It works only when configuration is added in all build.gradle files of child modules.

File: dependency-check-suppressions.xml [hunk 4]
```diff
@@ -106,8 +107,20 @@
...
     </suppress>
+
     <suppress>
-        <notes>cpe:2.3:a:google:guava:27.0 suppression, root cause and fix needs to be carried out separately</notes>
+        <notes>Launch-darkley jar uses guava@28.2-jre as an internal library and it is not exposed outside. Since it cannot be upgraded,
+            this supression is added. </notes>
         <cve>CVE-2020-8908</cve>
     </suppress>
+
+    <suppress until="2021-03-01">
```
Comment (AyanDuttaHMCTS) [inline dependency-check-suppressions.xml:108]: Is this date recommended by RPE?  By specifying a certain date for suppression, how confident we are that some latest Library would be available with fix to the CVEs

Comment (Jennings-HMCTS) [reply dependency-check-suppressions.xml:108]: This specific date is added based on other repositories where this supression has been added

**Review by aravindan89 — APPROVED** [review]

File: api-contract/build.gradle (hunk 1 above)
Comment (AyanDuttaHMCTS) [reply api-contract/build.gradle:3]: As discussed & shown please keep the config in Root Build.gradle [all projects section]

File: api/build.gradle (hunk 2 above)
Comment (AyanDuttaHMCTS) [reply api/build.gradle:37]: As discussed & shown please keep the config in Root Build.gradle [all projects section]

File: dependency-check-suppressions.xml (hunk 4 above)
Comment (AyanDuttaHMCTS) [reply dependency-check-suppressions.xml:108]: This way the build will go fail on 2nd March 2021

File: gov-pay-client/build.gradle [hunk 5]
```diff
@@ -1,5 +1,32 @@
 description = ''
+configurations.all {
```
Comment (AyanDuttaHMCTS) [inline gov-pay-client/build.gradle:2]: As discussed & shown please keep the config in Root Build.gradle [all projects section]

File: model/build.gradle [hunk 6]
```diff
@@ -1,5 +1,33 @@
 
 description = ''
+configurations.all {
```
Comment (AyanDuttaHMCTS) [inline model/build.gradle:3]: As discussed & shown please keep the config in Root Build.gradle [all projects section]

File: model/build.gradle [hunk 7]
```diff
@@ -12,7 +40,7 @@ dependencies {
     compile group: 'commons-validator', name: 'commons-validator', version:'1.6'
     compile group: 'uk.gov.hmcts.reform', name: 'logging-appinsights', version: '5.1.1'
     compile group: 'com.netflix.hystrix', name: 'hystrix-javanica', version: '1.5.12'
-    compile group: 'com.launchdarkly', name: 'launchdarkly-java-server-sdk', version: '5.0.0'
+    compile group: 'com.launchdarkly', name: 'launchdarkly-java-server-sdk', version: '5.2.1'
```
Comment (AyanDuttaHMCTS) [inline model/build.gradle:43]: is this still required ?

File: otp/build.gradle [hunk 8]
```diff
@@ -1,4 +1,32 @@
 description = 'OTP (One Time Password) utility in Java'
+configurations.all {
```
Comment (AyanDuttaHMCTS) [inline otp/build.gradle:2]: As discussed & shown please keep the config in Root Build.gradle [all projects section]

File: reference-data/build.gradle [hunk 9]
```diff
@@ -1,5 +1,32 @@
 description = 'Managing reference data required for payments'
+configurations.all {
```
Comment (AyanDuttaHMCTS) [inline reference-data/build.gradle:2]: As discussed & shown please keep the config in Root Build.gradle [all projects section]

**Review by AnilDhawaleGitHub — APPROVED** [review]

File: dependency-check-suppressions.xml (hunk 4 above)
Comment (AyanDuttaHMCTS) [reply dependency-check-suppressions.xml:108]: As discussed with @YasharRahvar, let's have 2 weeks of buffer date for a CVE to be rechecked for it's remediation. So the a suppression on 1st of Jan,2021 would have a date of 15th Jan, 2021

**Review by AyanDuttaHMCTS — APPROVED** [review]

**Pull request merged**